allennlp/data/token_indexers/token_characters_indexer.py

```diff
@@ -38,7 +38,7 @@ class TokenCharactersIndexer(TokenIndexer[List[int]]):
     # pylint: disable=no-self-use
     def __init__(self,
                  namespace: str = 'token_characters',
-                 character_tokenizer: CharacterTokenizer = CharacterTokenizer(),
+                 character_tokenizer: Tokenizer = CharacterTokenizer(),
```

Comment:
this is failing tests because of a bug in the automatic `FromParams`. Basically it's not handling default values correctly in the case where the type is itself `from_params`. Let me see if I can fix that.

Reply:
wait, I take that back, the `FromParams` code is right, let me look again and see why this is failing tests